config/remote/constants.go

```diff
@@ -55,51 +55,59 @@ func (e EventType) String() string {
 type DownloadErrorKind int
 
 const (
-	DownloadErrorRemoteHashNotFound DownloadErrorKind = iota
-	DownloadErrorRemoteFileNotFound
+	DownloadErrorFileDownload DownloadErrorKind = iota
+	DownloadErrorFileRename
+	DownloadErrorHashIntegrity
+	DownloadErrorHashParsing
+	DownloadErrorIncludeFile
 	DownloadErrorIntegrity
-	DownloadErrorFileDownload
+	DownloadErrorLocalFS
 	DownloadErrorNetwork
 	DownloadErrorOther
+	DownloadErrorParsing
+	DownloadErrorRemoteFileNotFound
+	DownloadErrorRemoteHashNotFound
+	DownloadErrorWriteHash
+	DownloadErrorWriteJson
 )
 
 func (e DownloadErrorKind) String() string {
 	switch e {
-	case DownloadErrorRemoteHashNotFound:
-		return "remote_hash_not_found"
-	case DownloadErrorRemoteFileNotFound:
-		return "remote_file_not_found"
-	case DownloadErrorIntegrity:
-		return "integrity_error"
 	case DownloadErrorFileDownload:
 		return "file_download_error"
+	case DownloadErrorFileRename:
+		return "file_rename_error"
+	case DownloadErrorHashIntegrity:
+		return "hash_integrity_error"
+	case DownloadErrorHashParsing:
+		return "hash_parsing_error"
+	case DownloadErrorIncludeFile:
+		return "include_file_error"
+	case DownloadErrorIntegrity:
+		return "integrity_error"
+	case DownloadErrorLocalFS:
+		return "local_fs_error"
 	case DownloadErrorNetwork:
 		return "network_error"
 	case DownloadErrorOther:
 		return "other_error"
+	case DownloadErrorParsing:
+		return "parsing_error"
+	case DownloadErrorRemoteFileNotFound:
+		return "remote_file_not_found"
+	case DownloadErrorRemoteHashNotFound:
+		return "remote_hash_not_found"
+	case DownloadErrorWriteHash:
+		return "write_hash_error"
+	case DownloadErrorWriteJson:
+		return "write_json_error"
```

Comment:
Wouldn't be easier to put those in a map and then get the string from the map based on the kind?

Reply:
Main reason why I chose switch, is a potential extension to the DownloadErrorKind. With a leverage of static-code analysis tool we would be able to easily spot a not handled case. Which might not be the case in map-bases approach. Not a big advantage, but still a small step toward code hygiene. :wink: 

If however you still would like to switch into map-based approach, then I'm perfectly happy to change that.